ballerina/tests/mock_service.bal

```diff
@@ -0,0 +1,219 @@
+// Copyright (c) 2025, WSO2 LLC. (http://www.wso2.com).
+//
+// WSO2 LLC. licenses this file to you under the Apache License,
+// Version 2.0 (the "License"); you may not use this file except
+// in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+import ballerina/http;
+
+final purchase_unit[] & readonly pUnits = [
+    {
+        reference_id: "default",
+        amount: {
+            currency_code: "USD",
+            value: "200.00",
+            breakdown: {
+                item_total: {
+                    currency_code: "USD",
+                    value: "180.00"
+                },
+                shipping: {
+                    currency_code: "USD",
+                    value: "20.00"
+                }
+            }
+        },
+        payee: {
+            email_address: "sb-wrafi36853704@business.example.com",
+            merchant_id: "M6V5C6A45Z32W"
+        }
+    }
+];
+
+final 'order & readonly sampleCaptureOrder = {
+    id: "6JE657202M751084B",
+    intent: "CAPTURE",
+    status: "CREATED",
+    purchase_units: pUnits,
+    create_time: "2025-06-17T08:53:19Z"
+};
+
+final 'order & readonly sampleAuthorizeOrder = {
+    id: "6JE657202M751084C",
+    intent: "AUTHORIZE",
+    status: "CREATED",
+    purchase_units: pUnits,
+    create_time: "2025-06-17T08:53:19Z"
+};
+
+service on new http:Listener(9090) {
+    resource isolated function post orders(@http:Payload order_request payload) returns 'order|http:NotFound {

```

Comment:
No need to have the payload annotation if the parameter is a structural type

[1] https://ballerina.io/learn/by-example/http-service-data-binding/

```suggestion
    resource isolated function post orders(order_request payload) returns 'order|http:NotFound {

```